# Week 3 Variant Calling -- Feedback

1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 = 10 points out of 10 possible points

1. Index genome

  ~* what code did you use? --> +0~
  * *in `livecoding.md` --> +1*

2. align reads

  ~* you're missing a sample in your `for` loop (A01_09)~
  * *sample not missing, just ran separately as noted in `livecoding.md`*


3. sort bam files and index sorted bam files (0.5 points each)

  * good use of the single `for` loop --> +1

4. variant call with freebayes

  ~* what code did you use? --> +0~
  * *very good! in `livecoding.md`--> +1*

5. filter variants

  ~* what code did you use? --> +0~
  * *good, in `livecoding.md` --> +1*

6. decompose complex haplotypes

  ~* what code did you use? --> +0~
  * *good, in `livecoding.md` --> +1*

7. variant effect prediction

  ~* what code did you use? --> +0~
  * *good, in `livecoding.md` --> +1*

8. python plotting script

  * very good and nice list comprehension in converting the values to integers/floats. You could also use the float and int functions when you append the values to the lists in the first place --> +1

9. 4 panel plot (0.25 points each panel)

  * good plot --> +1
  * You can try to rotate the tick labels for the variant effects so that they don't overlap, something like `plt.xticks(rotation = 45)`

10. 1000 line vcf

  * --> +1

updated feedback in *italics*
